### PR TITLE
[SPARK-5914] to run spark-submit requiring only user perm on windows

### PR DESCRIPTION
**Pull request by judynash**
Because windows on-default does not grant read permission to jars except to admin, spark-submit would fail with "ClassNotFound" exception if user runs slave service with only user permission. 
This fix is to add read permission to owner of the jar (which would be the slave service account in windows )

### CODE REVIEW

File: core/src/main/scala/org/apache/spark/util/Utils.scala [hunk 1]
```diff
@@ -439,7 +439,14 @@ private[spark] object Utils extends Logging {
       executeAndGetOutput(Seq("tar", "-xf", fileName), targetDir)
     }
     // Make the file executable - That's necessary for scripts
-    FileUtil.chmod(targetFile.getAbsolutePath, "a+x")
+    if(isWindows){
```
Comment (pwendell) [inline core/src/main/scala/org/apache/spark/util/Utils.scala]: mind adding a space after `if`?

**Comment by pwendell** [comment]
Jenkins, test this please.

File: core/src/main/scala/org/apache/spark/util/Utils.scala [hunk 2]
```diff
@@ -439,7 +439,14 @@ private[spark] object Utils extends Logging {
...
+      // Windows does not grant read permission by default to non-admin users
+      // Add read permission to owner explicitly
+      FileUtil.chmod(targetFile.getAbsolutePath, "u+r")
+      FileUtil.chmod(targetFile.getAbsolutePath, "a+x")
+    } else {
+      FileUtil.chmod(targetFile.getAbsolutePath, "a+x")
```
Comment (srowen) [inline core/src/main/scala/org/apache/spark/util/Utils.scala]: This doesn't need to be in both branches of the `if`  right -- this boils down to just adding `u+r` if `isWindows` and not affecting the current line.

**Comment by judynash** [comment]
Looks like the test didn't rerun after submitting updated code?

**Comment by srowen** [comment]
Jenkins, retest this please

**Comment by srowen** [comment]
LGTM. I'll fix the space before the brace on merge. This seems like a clean, simple fix.

**Pull request closed without merging**